src/file_part.rs

```diff
@@ -0,0 +1,180 @@
+fn compare_json_values(value1: &Value, value2: &Value) -> bool {
+    match (value1, value2) {
+        (&Value::Object(ref obj1), &Value::Object(ref obj2)) => {
+            if obj1.len() != obj2.len() {
+                return false;
+            }
+            for (key, value) in obj1 {
+                if !obj2.contains_key(key) || !compare_json_values(value, &obj2[key]) {
+                    return false;
+                }
+            }
+            true
+        }
+        (&Value::Array(ref arr1), &Value::Array(ref arr2)) => {
+            if arr1.len() != arr2.len() {
+                return false;
+            }
+            arr1.iter()
+                .zip(arr2.iter())
+                .all(|(v1, v2)| compare_json_values(v1, v2))
+        }
+        _ => value1 == value2,
+    }
+}
+
+/// Compares the content of two JSON files.
+///
+/// This function compares the content of two JSON files. It returns `Ok(true)` if the files are
+/// equal, `Ok(false)` if they are different, and an error message if any error occurs during
+/// the comparison.
+fn compare_json_files(file1: &str, file2: &str) -> Result<bool, String> {
+    info!("Comparing JSON files: {} and {}", file1, file2);
+    let content1 = read_file_content(file1)
+        .map_err(|err| format!("Failed to read file {}: {}", file1, err))?;
+    let content2 = read_file_content(file2)
+        .map_err(|err| format!("Failed to read file {}: {}", file2, err))?;
+    let json_value1 = parse_json(&content1)
+        .map_err(|err| format!("Failed to parse JSON from file {}: {}", file1, err))?;
+    let json_value2 = parse_json(&content2)
+        .map_err(|err| format!("Failed to parse JSON from file {}: {}", file2, err))?;
+    info!("Comparison complete.");
+    Ok(compare_json_values(&json_value1, &json_value2))
+}
+
+/// Performs file-related tasks.
+///
+/// This function performs file-related tasks including reading headlines from a web page,
+/// writing them to a JSON file, comparing JSON files, removing and renaming files, and sending
+/// updates via Telegram. It logs information about each step and any errors encountered.
+pub async fn file_work(url: &str) {
+    info!("Starting file work...");
+    let headlines = read_page_to_json_str_headlines(url)
+        .await
+        .expect("Failed to read headlines from page");
+    write_headlines_to_json_file(headlines)
+        .await
+        .expect("Failed to write headlines to JSON file");
+
+    match compare_json_files(FILE1, FILE2) {
+        Ok(true) => info!("The JSON files are equal. Nothing new."),
+        Ok(false) => {
+            info!("The JSON files are different.");
+
+            if let Err(err) = fs::remove_file(FILE2) {
+                error!("Failed to remove file {}: {}", FILE2, err);
+            }
+            if let Err(err) = fs::rename(FILE1, FILE2) {
+                error!("Failed to rename file {}: {}", FILE1, err);
+            }
+            send_first_upd().await;
```

Comment:
I don't quite understand this.
`main` calls `file_work` in a loop, which calls `send_first_upd` in case the files were changed,
and `send_first_upd` starts `teloxide::repl` `Bot`? That doesn't seem correct.
The bot repl should run only one instance, and this code spawns a new REPL each time it sees updates?